src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -0,0 +1,106 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=too-few-public-methods,too-many-arguments,no-self-use,too-many-locals,line-too-long
+
+import shlex
+from azure.cli.core.util import CLIError
+from azure.mgmt.container.models import (ContainerGroup, Container, ContainerPort, Port, IpAddress,
+                                         EnvironmentVariable, ImageRegistryCredential, ResourceRequirements,
+                                         ResourceRequests, ContainerGroupNetworkProtocol, OperatingSystemTypes)
+
+ACR_SERVER_SUFFIX = ".azurecr.io/"
+
+def list_all_containers(client):
+    """List all container groups. """
+    return client.container_groups.list_all()
+
+def list_containers(client, resource_group_name):
+    """List all container groups in a resource group. """
+    return client.container_groups.list(resource_group_name)
+
+def get_container(client, resource_group_name, name):
+    """Show details of a container group. """
+    return client.container_groups.get(resource_group_name, name)
+
+def delete_container(client, resource_group_name, name, **kwargs):
+    """Delete a container group. """
+    return client.container_groups.delete(resource_group_name, name)
+
+def create_container(client,
+                     resource_group_name,
+                     name,
+                     image,
+                     location=None,
+                     cpu=1,
+                     memory=1.5,
+                     port=80,
+                     os_type='Linux',
+                     ip_address=None,
+                     command_line=None,
+                     environment_variables=None,
+                     image_registry_login_server=None,
+                     image_registry_username=None,
+                     image_registry_password=None):
+    """"Create a container group. """
+
+    container_resource_requirements = None
+    if cpu is not None or memory is not None:
+        container_resource_requests = ResourceRequests(memory_in_gb=memory, cpu=cpu)
+        container_resource_requirements = ResourceRequirements(requests=container_resource_requests)
+
+    image_registry_credentials = None
+    if image_registry_login_server is not None:
+        if image_registry_username is None:
+            raise CLIError('Missing --image-registry-username.')
+        if image_registry_password is None:
```

Comment:
You should support using the in-built prompting for this password if it is not supplied but is required. Putting password as command line arguments is a security risk because it gets logged by the OS.